lib/util/batch.js

```diff
@@ -121,7 +121,25 @@ function supportsCql31(version) {
   return version.major > 3 || (version.major === 3 && version.minor >= 1);
 }
 
-var endsWithSemicolonRegex = new RegExp(/\s*;\s*$/);
+function containsDeleteOrUpdateStatement(query) {
+  var match = false;
+  if (query instanceof Batch) {
```

Comment:
Awesome. I wouldn't block this PR on that however since as I said it's done in many other places. I'll open an issue so it can be addressed separately.